wordpress-mobile/WordPress-Android
# Social sharing adding tracking for post settings

This PR adds a tracker for the JP social improvements feature.

Tracked events are:

**JETPACK_SOCIAL_AUTO_SHARING_CONNECTION_TOGGLED**
triggered when a user toggle the switch to share/not-share to a social network

example: `Tracked: jetpack_social_auto_sharing_connection_toggled, Properties: {"source":"post_settings","value":false}`

**JETPACK_SOCIAL_SHARE_LIMIT_DISPLAYED**
triggered when the remaining number of share message is shown

example: `Tracked: jetpack_social_share_limit_displayed, Properties: {"source":"post_settings"}`

**JETPACK_SOCIAL_UPGRADE_LINK_TAPPED**
triggered when the user taps on "Subscribe to share more"

example: `Tracked: jetpack_social_upgrade_link_tapped, Properties: {"source":"post_settings"}`

**JETPACK_SOCIAL_ADD_CONNECTION_CTA_DISPLAYED**
triggered when the no connections screen is shown

example: `Tracked: jetpack_social_add_connection_cta_displayed, Properties: {"source":"post_settings"}`

**JETPACK_SOCIAL_ADD_CONNECTION_TAPPED**        
triggered when the user tap on "Connect accounts"

example: `Tracked: jetpack_social_add_connection_tapped, Properties: {"source":"post_settings"}`

**JETPACK_SOCIAL_ADD_CONNECTION_DISMISSED**
triggered when the user tap on "Not now"

example: `Tracked: jetpack_social_add_connection_dismissed, Properties: {"source":"post_settings"}`

NOTE: in the above events the source parameter can be `post_settings` or `pre_publishing` depending on if they are triggered from the post settings or the pre-publishing sheet.

To test:
- Check the events are triggered as described above from the post settings
- Check the events are triggered as described above from the pre-publishing sheet

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests are in place for regressions; dedicated unit tests will be added in a separate PR.

PR submission checklist:


UI Changes testing checklist:
